### ISSUE TITLE
Improvements to Mappers.java

### ISSUE THREAD

**Issue by Petrychyn** [mapstruct/mapstruct#721]
As described here:

https://groups.google.com/forum/?fromgroups=#!topic/mapstruct-users/Wkoz8KEKTYA

The loading process in Mappers.java should attempt to use 
```java
clazz.getClassLoader();
```
before failing on line 94 in the case where the current thread's contextClassLoader cannot load the class.

There is also room to add additional validation to this method as described in the comments beginning on line 66.

```java
// Check that
// - clazz is an interface
// - the implementation type implements clazz
// - clazz is annotated with @Mapper
```

**Comment by agudian** [comment]
Great, would you like to give it a shot and prepare pull-request?

The check for `clazz` being an interface doesn't need  to be done, btw. That comment is from a time where we didn't yet support abstract classes to be annotated with `@Mapper`... :wink:

**Comment by gunnarmorling** [comment]
Indeed. I thought about checking whether the given type actually is `@Mapper`-annotated, but I don't think it's worth to change the annotation's retention just for that.

**Comment by gunnarmorling** [comment]
This issue has been fixed on master as of #801. I'm going to close this one.
